cadcdata/cadcdata/tests/test_inventory.py

```diff
@@ -0,0 +1,571 @@
+    post_mock.reset_mock()
+    client._get_transfer_urls = Mock(return_value=['url1'])
+    client.cadcinfo = Mock(
+        return_value=cadcdata.FileInfo('cadc:TEST/putfile',
+                                       file_type='text/plain',
+                                       encoding='us-ascii',
+                                       md5sum='0x123456789'))
+    client.cadcput('cadc:TEST/putfile', file_name, replace=True,
+                   file_type='text/plain', file_encoding='us-ascii',
+                   md5_checksum='0x123456789')
+    put_mock.assert_not_called()
+    post_mock.assert_not_called()
+
+    # replace non existing file
+    client.cadcinfo.side_effect = [exceptions.NotFoundException()]
+    with pytest.raises(AttributeError):
+        client.cadcput('cadc:TEST/putfile', file_name, replace=True)
+
+    # put a new file that already exists
+    client.cadcinfo = Mock(
+        return_value=cadcdata.FileInfo('cadc:TEST/putfile',
+                                       file_type='text/plain',
+                                       encoding='us-ascii',
+                                       md5sum='0x123456789'))
+    with pytest.raises(AttributeError):
+        client.cadcput('cadc:TEST/putfile', file_name, replace=False)
+
+
+@patch('cadcdata.core.net.BaseWsClient')
+def test_remove(basews_mock):
+    client = StorageInventoryClient(auth.Subject())
+    # test a put
+    remove_mock = Mock()
+    basews_mock.return_value.delete = remove_mock
+    with pytest.raises(exceptions.UnauthorizedException):
+        client.cadcremove('cadc:TEST/removefile')
+    client._cadc_client.subject.anon = False  # authenticate the user
+
+    client._get_transfer_urls = Mock(return_value=['url1'])
+    client.cadcremove('cadc:TEST/removefile')
+    remove_mock.assert_called_with('url1')
+
+
+@patch('cadcdata.storageinv.net.BaseWsClient')
+def test_info(basews_mock):
+    client = StorageInventoryClient(auth.Subject())
+    # test an info
+    file_name = 'myfile.txt'
+    id = 'cadc:TEST/myfile.txt'
+    size = '123'
+    md5sum = '0x123'
+    file_type = 'txt'
+    file_encoding = 'gzip'
+    lastmod = 'Mon, 17 May 2021 09:49:18 GMT'
+
+    h = CaseInsensitiveDict()
+    h['Content-Disposition'] = 'inline; filename={}'.format(file_name)
+    h['Content-Length'] = size
+    h['Digest'] = 'md5={}'.format(
+        base64.b64encode(md5sum.encode('ascii')).decode('ascii'))
+    h['Content-Type'] = file_type
+    h['Content-Encoding'] = file_encoding
+    h['Last-Modified'] = lastmod
+    response = Mock()
+    response.headers = h
+    basews_mock.return_value.head.return_value = response
+    info = client.cadcinfo(id)
+    assert id == info.id
+    assert file_name == info.name
+    assert int(size) == info.size
+    assert md5sum == info.md5sum
+    assert file_type == info.file_type
+    assert file_encoding == info.encoding
+    assert datetime.datetime.strptime(
+        lastmod, '%a, %d %b %Y %I:%M:%S %Z') == info.lastmod
+
```

Comment:
Possible test - call for a non-existent file?

Reply:
Added